respx/router.py

```diff
@@ -154,6 +154,14 @@ def request(
         name: Optional[str] = None,
         **lookups: Any,
     ) -> Route:
+        if lookups:
+            # Validate that lookups doesn't contain method or url
+            pattern_keys = {p.split("__")[0] for p in lookups.keys()}
+            if "method" in pattern_keys:
+                raise TypeError("Got multiple values for pattern 'method'")
+            if url and "url" in pattern_keys:
```

Comment:
This could be an `elif` since we won't continue when first statement evaluates to true.

```suggestion
            elif url and "url" in pattern_keys:
```